.evergreen/scripts/integration-tests.sh

```diff
@@ -7,11 +7,9 @@
 # MONGODB_VERSION: latest, 4.2, 4.0
 # TOPOLOGY: server, replica_set, sharded_cluster
 # AUTH: auth, noauth
-# AUTHSOURCE
 # IPV4_ONLY: off, on
 # SSL: openssl, darwinssl, winssl, nossl
-# ORCHESTRATION_FILE: <file name in orchestration_configs/ without .json>
-#   If this is not set, the file name is constructed from other options.
+# ORCHESTRATION_FILE: <file name in DET configs/${TOPOLOGY}s/>
```

Comment:
The C driver contains orchestration configs that are not in DET:
- `orchestration_configs/servers/basic-ipv4-only.json`
- `orchestration_configs/servers/snappy.json`
- `orchestration_configs/servers/zstd.json`

On further look, I do not the extra configs provide much meaningful test coverage.

`snappy` and `zstd` may not be relevant for modern servers. The default [enables all three](https://www.mongodb.com/docs/manual/reference/program/mongod/#std-option-mongod.--networkMessageCompressors).

Tasks specifying `IPV4_ONLY` (to use `basic-ipv4-only`) also [specify the URI `mongodb://127.0.0.1`](https://github.com/mongodb/mongo-c-driver/blob/56cbe883be2fe84812e35ea8fae181dc7e0be8ce/.evergreen/generated_configs/legacy-config.yml#L1706-L1707), so I expect those tasks connect with IPv4 without extra server configuration.

Suggest:
- Remove C driver `orchestration_configs`.
- Remove mention of `IPV4_ONLY`.

Reply:
Thank you. Done as suggested.

I preserved the `MONGOC_CHECK_IPV6` env var (formerly conditioned on `"$IPV4_ONLY" != "on"`; now unconditionally defined in EVG scripts) in case there are downstream users that still depend on it. LMK if you would like this removed from the test suite as well.

Reply:
I slightly prefer removing `MONGOC_CHECK_IPV6`. I do not think preserving test environment variables is needed, and I expect removing `MONGOC_CHECK_IPV6` would not be disruptive.

Reply:
Done.